src/main/java/org/mule/extension/api/introspection/metadata/MetadataResolverFactory.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extension.api.introspection.metadata;
+
+import org.mule.api.metadata.resolving.MetadataContentResolver;
+import org.mule.api.metadata.resolving.MetadataKeysResolver;
+import org.mule.api.metadata.resolving.MetadataOutputResolver;
+
+/**
+ * Provides instances of the {@link MetadataKeysResolver}, {@link MetadataKeysResolver}
+ * and {@link MetadataOutputResolver} resolving types associated to a Component
+ *
+ * @since 1.0
+ */
+public interface MetadataResolverFactory
```

Comment:
do instances of this class always return the same instances or new ones? If the latter, then replace the get prefix with create